File: LeetcodeJava/src/Num125.java
import java.util.Locale;

public class Num125 {
    public static void main(String[] args) {
        String s = "A man, a plan, a canal: Panama";
//        String s = "race a car";
//        String s = " ";
        System.out.println(isPalindrome(s));
    }

    public static boolean isPalindrome(String s) {
        String alteredString = s.toLowerCase(Locale.ROOT);
        alteredString = alteredString.replaceAll("[^a-zA-Z0-9]", "");
        System.out.println(alteredString);
        return alteredString.equals(s);
    }
}

File: Leetcode/src/main/kotlin/Num125.kt
fun main(args: Array<String>) {
//    val s = "A man, a plan, a canal: Panama"
    val s = "race a car"
//    val s = " "
    println(isPalindrome(s))
}

fun isPalindrome(s: String): Boolean {
    val alteredString = s.replace("[^a-zA-Z0-9]".toRegex(), "").toLowerCase()
    val reversed = alteredString.reversed()
    return reversed == alteredString
}